iocore/net/TLSSessionResumptionSupport.cc

```diff
@@ -172,6 +172,30 @@ TLSSessionResumptionSupport::getSession(SSL *ssl, const unsigned char *id, int l
   return session;
 }
 
+SSL_SESSION *
+TLSSessionResumptionSupport::getOriginSession(SSL *ssl, const std::string &lookup_key)
```

Comment:
I'm OK with adding this for now, but I wonder if we can have only one `getSession` and lookup different caches inside it depends on the connection direction (SSL_is_server). I don't think we never call `getOriginSession` on client sessions (inbound connections), and vice versa.